src/snowflake/cli/plugins/nativeapp/codegen/snowpark/extension_function_utils.py

```diff
@@ -0,0 +1,368 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    Union,
+)
+
+from click.exceptions import ClickException
+
+
+class MalformedExtensionFunctionError(ClickException):
+    """Required extension function attribute is missing."""
+
+    def __init__(self, message: str):
+        super().__init__(message=message)
+
+
+# This prefix is created by Snowpark for an extension function when the user has not supplied any themselves.
+# TODO: move to sandbox execution to omit object name in this case: https://github.com/snowflakedb/snowflake-cli/pull/1056/files#r1599784063
+TEMP_OBJECT_NAME_PREFIX = "SNOWPARK_TEMP_"
+
+
+def get_object_type_as_text(name: str) -> str:
+    """
+    Replace underscores with spaces in a given string.
+
+    Parameters:
+        name (str): Any arbitrary string
+    Returns:
+        A string that has replaced underscores with spaces.
+    """
+    return name.replace("_", " ")
+
+
+def _sanitize_str_attribute(
+    ex_fn: Dict[str, Any],
```

Comment:
I see `ex_fn` a lot but I'm really not sure what it conceptually maps to. Is this like a property bag for a sproc/udf? Maybe we can use a type alias and add a docstring there?

Reply:
yeah, it's an extension function property bag expressed as a dict. Rather temporary I'd say, the validation piece here is very difficult to reason about already.